io.zeebe.broker.it.job.ActivateJobsTest.shouldActivateJobsOnPartitionsRoundRobin is flaky

```
Error Message
Expecting:
  <[0, 1, 0]>
to contain only:
  <[0, 1, 2]>
but could not find the following elements:
  <[2]>
Stacktrace
java.lang.AssertionError: 

Expecting:
  <[0, 1, 0]>
to contain only:
  <[0, 1, 2]>
but could not find the following elements:
  <[2]>

	at io.zeebe.broker.it.job.ActivateJobsTest.shouldActivateJobsOnPartitionsRoundRobin(ActivateJobsTest.java:172)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/qa/integration-tests/src/test/java/io/zeebe/broker/it/job/ActivateJobsTest.java
+++ b/qa/integration-tests/src/test/java/io/zeebe/broker/it/job/ActivateJobsTest.java
@@ -16,6 +16,7 @@
 package io.zeebe.broker.it.job;
 
 import static io.zeebe.broker.test.EmbeddedBrokerConfigurator.setPartitionCount;
+import static io.zeebe.test.util.TestUtil.waitUntil;
 import static io.zeebe.test.util.record.RecordingExporter.jobRecords;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.tuple;
@@ -23,13 +24,13 @@ import static org.assertj.core.api.Assertions.tuple;
 import io.zeebe.broker.it.GrpcClientRule;
 import io.zeebe.broker.test.EmbeddedBrokerRule;
 import io.zeebe.client.ZeebeClient;
+import io.zeebe.client.api.commands.PartitionInfo;
 import io.zeebe.client.api.response.ActivateJobsResponse;
 import io.zeebe.client.api.response.ActivatedJob;
 import io.zeebe.exporter.api.record.Assertions;
 import io.zeebe.protocol.Protocol;
 import io.zeebe.protocol.intent.JobBatchIntent;
 import io.zeebe.protocol.intent.JobIntent;
-import io.zeebe.test.util.TestUtil;
 import io.zeebe.test.util.record.RecordingExporter;
 import io.zeebe.util.StreamUtil;
 import io.zeebe.util.collection.Tuple;
@@ -70,7 +71,15 @@ public class ActivateJobsTest {
 
   @Before
   public void setUp() {
+
     client = clientRule.getClient();
+    waitUntil(
+        () ->
+            client.newTopologyRequest().send().join().getBrokers().stream()
+                    .flatMap(b -> b.getPartitions().stream())
+                    .filter(PartitionInfo::isLeader)
+                    .count()
+                == PARTITION_COUNT);
   }
 
   @After
@@ -100,7 +109,7 @@ public class ActivateJobsTest {
             .send()
             .join();
 
-    TestUtil.waitUntil(() -> jobRecords(JobIntent.ACTIVATED).limit(amount).count() == amount);
+    waitUntil(() -> jobRecords(JobIntent.ACTIVATED).limit(amount).count() == amount);
 
     // then
     assertThat(response.getJobs()).extracting(ActivatedJob::getKey).containsOnlyElementsOf(jobKeys);

